src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -143,7 +143,7 @@ SELECT CASE WHEN (count(c) > 0) THEN true ELSE false END
     @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.lectureUnits" })
     Optional<Course> findWithEagerLecturesAndLectureUnitsById(long courseId);
 
-    @EntityGraph(type = LOAD, attributePaths = { "organizations", "competencies", "prerequisites", "tutorialGroupsConfiguration", "onlineCourseConfiguration" })
+    @EntityGraph(type = LOAD, attributePaths = { "organizations", "competencies", "prerequisites", "tutorialGroupsConfiguration", "onlineCourseConfiguration", "irisSettings" })
     Optional<Course> findWithEagerOrganizationsAndCompetenciesAndOnlineConfigurationById(long courseId);
```

Comment:
This loads a lot more than the name suggests. Can you maybe rename this to `findCourseForUpdate`?

Reply:
Done